test/e2e/utils.go

```diff
@@ -73,3 +79,65 @@ func getThanosRules(body []byte, expGroupName, expRuleName string) error {
 	}
 	return fmt.Errorf("'%s' alert not found in '%s' group", expRuleName, expGroupName)
 }
+
+func createSelfSignedMTLSArtifacts(s *v1.Secret) error {
+	newCAConfig, err := crypto.MakeSelfSignedCAConfig(
+		fmt.Sprintf("%s@%d", "openshift-cluster-monitoring-test", time.Now().Unix()),
+		crypto.DefaultCertificateLifetimeInDays,
+	)
+	if err != nil {
+		return errors.Wrap(err, "error generating self signed CA")
+	}
+
+	newCA := &crypto.CA{
+		SerialGenerator: &crypto.RandomSerialGenerator{},
+		Config:          newCAConfig,
+	}
+
+	newCABytes, newCAKeyBytes, err := newCA.Config.GetPEMBytes()
+	if err != nil {
+		return errors.Wrap(err, "error getting PEM bytes from CA")
+	}
+
+	s.Data["ca.crt"] = newCABytes
+	s.Data["ca.key"] = newCAKeyBytes
+	// create serving certs maybe according to secret keys?
+	{
+		cfg, err := newCA.MakeServerCert(
+			sets.NewString(string(s.Data["serving-cert-url"])),
+			crypto.DefaultCertificateLifetimeInDays,
+		)
+		if err != nil {
+			return errors.Wrap(err, "error making server certificate")
+		}
+
+		crt, key, err := cfg.GetPEMBytes()
+		if err != nil {
+			return errors.Wrap(err, "error getting PEM bytes for server certificate")
+		}
+		s.Data["server.crt"] = crt
```

Comment:
do we need this entry? IIUC only `server.key` and `server-ca.pem` are used.

Reply:
Yes `server-ca.pem` is mostly a convenience field for the sample app, as that expects the certificate and CA on one pem chunk. Since this lives in `utils` I left `server.key` in there for future use cases, I figured it wouldn't harm anything.

Reply:
:+1: